controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -1386,9 +1461,13 @@ func (r *BareMetalHostReconciler) manageHostPower(prov provisioner.Provisioner,
 		}
 	}
 
-	provState := info.host.Status.Provisioning.State
-	// Normal reboots only work in provisioned states, changing online is also possible for available hosts.
-	isProvisioned := provState == metal3api.StateProvisioned || provState == metal3api.StateExternallyProvisioned
+	servicingAllowed := isProvisioned && !info.host.Status.PoweredOn && desiredPowerOnState
+	if servicingAllowed || info.host.Status.OperationalStatus == metal3api.OperationalStatusServicing || info.host.Status.ErrorType == metal3api.ServicingError {
+		result, isServicing := r.checkServicing(prov, info)
+		if result != nil && (result.Dirty() || isServicing) {
+			return result
+		}
+	}
 
 	desiredReboot, desiredRebootMode := hasRebootAnnotation(info, !isProvisioned)
```

Comment:
It does not matter: in this case Ironic orchestrates the reboot itself (which is a problem of its own, but let's leave it for another day).

Reply:
I'm confused now, because line 1477 says that we only start the service mode if the power state is off but we wanted it to be on. AFAIK that only occurs during a hard reboot.

Reply:
Ah, you're probably right. No, soft reboot is totally fine and potentially even desired to prevent data loss. From that point on, Ironic will take care of the process.

Reply:
TODO: check and make sure it's soft poweroff not soft reboot.

Reply:
comment from Dmitry: We don't even have a provisioner call for rebooting - so it shouldn't be of concern.